### PR TITLE
Hide qual_note questions from Data Table and Single Submission Modal

### LINKED ISSUE #4619: Don't display `qual_note` in the Data Table

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

The `qual_note` Analysis Question type doesn't contain any useful data, so we don't display it in the Project Data UI.

## Notes

We have two places that are responsible for data displaying (one for Data Table and other for Single Submission Modal). I added an exclusion in both of these places.

## Related issues

Fixes #4619

### CODE REVIEW

File: jsapp/js/components/submissions/submissionUtils.ts [hunk 1]
```diff
@@ -108,16 +110,23 @@ export class DisplayResponse {
 
 /**
  * Returns a sorted object of transcript/translation keys
+ *
+ * Note: we omit returning `qual_note` questions.
  */
-export function sortAnalysisFormJsonKeys(additionalFields: Array<{source: string, dtpath: string}>) {
+function sortAnalysisFormJsonKeys(additionalFields: AnalysisFormJsonField[]) {
```
Comment (duvld) [inline jsapp/js/components/submissions/submissionUtils.ts:116]: Was this made before AnalysisFormJsonField was created/extended? There's a lot more mandatory fields than the parameter it replaced

File: jsapp/js/components/processing/analysis/constants.ts [hunk 2]
```diff
@@ -16,6 +16,10 @@ export type AnalysisQuestionType =
   | 'qual_tags'
   | 'qual_text';
 
+// We use it on other parts of code that is not related to analysis and is not
```
Comment (duvld) [inline jsapp/js/components/processing/analysis/constants.ts:19]: I'm not sure what this comment means, I feel like it should say something like "We don't need to show qualitative note questions outside of analysis"? Or maybe I'm misunderstanding

File: jsapp/js/assetUtils.ts [hunk 3]
```diff
@@ -520,21 +521,26 @@ export function injectSupplementalRowsIntoListOfRows(
   // on Back end, to build a list of columns grouped by source question
   const additionalFields = asset.analysis_form_json?.additional_fields || [];
   const extraColsBySource: Record<string, AnalysisFormJsonField[]> = {};
-  additionalFields.forEach((add_field: AnalysisFormJsonField) => {
-    const sourceName: string = add_field.source;
+  additionalFields.forEach((field: AnalysisFormJsonField) => {
+    // We don't want to display notes in the UI, so we omit it here:
```
Comment (duvld) [inline jsapp/js/assetUtils.ts:525]: Maybe a bit more verbose of a comment?

File: jsapp/js/components/submissions/submissionUtils.ts (hunk 1 above)
Comment (magicznyleszek) [reply jsapp/js/components/submissions/submissionUtils.ts:116]: I was following the code that is using that function. It's actually being used only in one place, in `getSubmissionDisplayData`, which is passing `asset.analysis_form_json?.additional_fields` into it, and this prop is typed as `additional_fields: AnalysisFormJsonField[];`

File: jsapp/js/components/processing/analysis/constants.ts (hunk 2 above)
Comment (magicznyleszek) [reply jsapp/js/components/processing/analysis/constants.ts]: I wanted to justify having a `const` here, while that `AnalysisQuestionType` should be more than enough. I rewrote the comment, but maybe the best way is to not have a comment at all here?

My thinking right now is that before TypeScript we had these long elaborate enums to ensure every piece of code is using the same strings. After moving some code to TS it become clear, that having a type is safe enough, and keeping const/enums for list of names of things is overdoing things.

File: jsapp/js/assetUtils.ts (hunk 3 above)
Comment (magicznyleszek) [reply jsapp/js/assetUtils.ts]: Rewrote it here and in `submissionUtils.ts` too

**Pull request merged**